swan-lake/learn-the-platform/ballerina-persist/persist-client-api.md

```diff
@@ -0,0 +1,245 @@
+```ballerina
+// Create a new workspace record
+WorkspaceInsert workspace = {id: "WK001", workspaceType: "Host Desk", locationCode: "B001"};
+string[]|error workspaceId = sClient->/workspaces.post([workspace]);
+
+// Get the workspace record with id WK001
+Workspace|error workspace = sClient->/workspaces/WK001;
+
+// Update the workspace record with id WK001
+Workspace|error updated = sClient->/workspaces/WK001.put({workspaceType: "Cubical"});
+
+// Delete the workspace record with id WK001
+Workspace|error deleted = sClient->/workspaces/WK001.delete();
+
+// Get all the workspace records
+stream<Workspace, error?> workspaces = sClient->/workspaces;
+```
+
+#### Relation Operations
+
+You can perform relation operations on the `Workspace` table in the data store using the client object, like this:
+
+Let's extend the `Workspace` entity type with a relation field `employees` as follows,
+```ballerina
+type Workspace record {|
+    readonly string id;
+    string workspaceType;
+    Building location; // 1-n relation (parent)
+    Employee? employees; // 1-1 relation (child)
+|};
+
+type Employee record {|
+    readonly string id;
+    string firstName;
+    string lastName;
+    string email;
+    float salary;
+    Workspace workspace; // 1-1 relation (parent)
+|};
+
+type Building record {|
+    readonly string code;
+    string name;
+    string address;
+    Workspace[] workspaces; // 1-n relation (child)
+|};
+```
+
+You can retrieve Workspace record with the related Employee record by using a custom target type as follows, This is equivalent to a SQL join operation between the `Workspace` and `Employee` tables.
+
+```ballerina
+// Get the workspace record with id WK001 with the related employee record
+WorkspaceWithEmployee|error workspace = sClient->/workspaces/WK001.get();
+
+// WorkspaceWithEmployee is a record with the following fields
+type WorkspaceWithEmployee record {|
+    readonly string id;
+    string workspaceType;
+    string locationCode
+    Employee employees; // 1-1 relation (child)
+|};
+```
+
+This same as write write the following SQL query to retrieve data from MySQL database,
+```sql
+SELECT * 
+FROM workspace w
+INNER JOIN employee e ON w.id = e.workspaceId
+WHERE w.workspaceId = 'WK001';
+```
+
+The persisted client will automatically perform the join operation and return the result.
+
+#### Projection Operations
+
+You can retrieve Workspace record with the related Employee record and the related Building record by using a custom target type. When specifying the target type, you can specify the target type with subset of fields of the entity types as follows, 
+
+This is equivalent to a SQL join operation between the `Workspace`, `Employee` and `Building` tables and also a SQL projection operation.
+
+```ballerina
+// Get the workspace record with id WK001 with the related employee record and the related building record
```

Comment:
```suggestion
// Get the `workspace` record with the ID `WK001` with the related `employee` record and the related `building` record.
```
Should the first letter be capital of the words within backticks? They need to be consistent throughout.